tests/provisioning/loadgen_test.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import os
+import unittest
+import subprocess
+from shlex import split
+import json
+import urllib.request
+
+from google.cloud.container_v1.services import cluster_manager
+
+class TestLoadGenerator(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        """ Get the location of GKE cluster for later queries """
+        cls.name = 'projects/{0}/locations/{1}/clusters/cloud-ops-sandbox-loadgenerator'.format(getProjectId(), getClusterZone())
+        # authenticate container cluster
+        command=('gcloud config set project {0}'.format(getProjectId()))
+        subprocess.run(split(command))
+        # set kubectl context to cloud-ops-sandbox-loadgenerator
+        command=('gcloud container clusters get-credentials cloud-ops-sandbox-loadgenerator --zone {0}'.format(getClusterZone()))
```

Comment:
This might be problem when your run e2e tests of various components involving kubernetes on the same CI agent.
For example, say you run loadgen e2e test wired against gke1 and hipstershop e2e test wired to gke2.
If you switch the context, it's switches it for all the sessions for the same user on the same VM/container, so by switching to gke1, you might affect tests that run in parallel and expect to run against gke2.
To prevent that, in your tests, you should specify _--context_ with every kubectl command.